File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    const arrLength = arr.length;
    const left = arrLength%2 ? 0 : 1;
    
    for(let i = left; i< arrLength; i+=2){
        arr[i] += n;
    }
    
    return arr;
}



function solution(arr, n) {
    if(arr.length%2){
        arr = arr.map((v,i) => i%2?v:v+n)
    }else{
        arr = arr.map((v,i) => i%2?v+n:v)
    }
    return arr;
}

/*
시간복잡도 : O(1) -> O(m) : m은 arr의 길이
공간복잡도 : O(1) -> O(m) : m은 arr의 길이 ; map은 새로운 배열을 리턴하기 때문에 길이에 따라 달라짐
*/

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    if len(arr) % 2 == 1:
        return [arr[i] + n if i % 2 == 0 else arr[i] for i in range(len(arr))]
    else:
        return [arr[i] + n if i % 2 == 1 else arr[i] for i in range(len(arr))]
    
# 시간복잡도 : O(n) ; n은 arr의 길이
## 리턴하는 배열 내의 for문이 arr의 길이만큼 돌기 때문에 
# 공간복잡도 : O(n) ; n은 arr의 길이
## 리턴하는 배열의 크기가 arr의 길이이기 때문에
